In [12]:
%load_ext sql
!pip install ipython-sql
!pip install pymysql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [2]:
%sql mysql://USERNAME:PASSWORD@HOST/DATABASE

'Connected: admin@sql_project'

## Exploratory Query 1

First exploratory SQL query: 

How did the interest and popularity of a music genre change over the past 5 years? (2015-2020)? We will use data from Google Trends API on 9 various music genres. 


To combine the 9 genre interest tables, we will create a VIEW so that we can run the query one time rather than 9 times for each genre. We will use UNION to merge the tables together.

In [6]:
%%sql
CREATE VIEW genre_interest AS
    SELECT `date`, interest, "alternative" AS genre
    FROM alternative_interest
    UNION
    SELECT `date`, interest, "folk" AS genre
    FROM folk_interest
    UNION
    SELECT `date`, interest, "jazz" AS genre
    FROM jazz_interest
    UNION
    SELECT `date`, interest, "pop" AS genre
    FROM pop_interest
    UNION
    SELECT `date`, interest, "rap" AS genre
    FROM rap_interest
    UNION
    SELECT `date`, interest, "reggae" AS genre
    FROM reggae_interest
    UNION
    SELECT `date`, interest, "rnb" AS genre
    FROM rnb_interest
    UNION
    SELECT `date`, interest, "rock" AS genre
    FROM rock_interest
    UNION
    SELECT `date`, interest, "soul" AS genre
    FROM soul_interest;


 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
0 rows affected.


[]

Confirming the view works as proposed:

In [8]:
%%sql
SELECT *
FROM genre_interest

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
2349 rows affected.


[('2015-05-03', 83, 'alternative'),
 ('2015-05-10', 86, 'alternative'),
 ('2015-05-17', 83, 'alternative'),
 ('2015-05-24', 84, 'alternative'),
 ('2015-05-31', 88, 'alternative'),
 ('2015-06-07', 81, 'alternative'),
 ('2015-06-14', 78, 'alternative'),
 ('2015-06-21', 81, 'alternative'),
 ('2015-06-28', 90, 'alternative'),
 ('2015-07-05', 89, 'alternative'),
 ('2015-07-12', 89, 'alternative'),
 ('2015-07-19', 100, 'alternative'),
 ('2015-07-26', 82, 'alternative'),
 ('2015-08-02', 87, 'alternative'),
 ('2015-08-09', 88, 'alternative'),
 ('2015-08-16', 87, 'alternative'),
 ('2015-08-23', 86, 'alternative'),
 ('2015-08-30', 84, 'alternative'),
 ('2015-09-06', 76, 'alternative'),
 ('2015-09-13', 84, 'alternative'),
 ('2015-09-20', 83, 'alternative'),
 ('2015-09-27', 78, 'alternative'),
 ('2015-10-04', 80, 'alternative'),
 ('2015-10-11', 79, 'alternative'),
 ('2015-10-18', 86, 'alternative'),
 ('2015-10-25', 83, 'alternative'),
 ('2015-11-01', 81, 'alternative'),
 ('2015-11-08', 74, 'alternative'),
 ('2015-11-15', 79, 'alternative'),
 ('2015-11-22', 76, 'alternative'),
 ('2015-11-29', 82, 'alternative'),
 ('2015-12-06', 86, 'alternative'),
 ('2015-12-13', 76, 'alternative'),
 ('2015-12-20', 84, 'alternative'),
 ('2015-12-27', 79, 'alternative'),
 ('2016-01-03', 78, 'alternative'),
 ('2016-01-10', 72, 'alternative'),
 ('2016-01-17', 76, 'alternative'),
 ('2016-01-24', 77, 'alternative'),
 ('2016-01-31', 79, 'alternative'),
 ('2016-02-07', 75, 'alternative'),
 ('2016-02-14', 81, 'alternative'),
 ('2016-02-21', 77, 'alternative'),
 ('2016-02-28', 75, 'alternative'),
 ('2016-03-06', 81, 'alternative'),
 ('2016-03-13', 78, 'alternative'),
 ('2016-03-20', 76, 'alternative'),
 ('2016-03-27', 88, 'alternative'),
 ('2016-04-03', 80, 'alternative'),
 ('2016-04-10', 74, 'alternative'),
 ('2016-04-17', 78, 'alternative'),
 ('2016-04-24', 78, 'alternative'),
 ('2016-05-01', 82, 'alternative'),
 ('2016-05-08', 69, 'alternative'),
 ('2016-05-15', 80, 'alternative'),
 ('2016-05-22', 74, 'alternative'),
 ('2016-05-29', 65, 'alternative'),
 ('2016-06-05', 69, 'alternative'),
 ('2016-06-12', 75, 'alternative'),
 ('2016-06-19', 78, 'alternative'),
 ('2016-06-26', 79, 'alternative'),
 ('2016-07-03', 70, 'alternative'),
 ('2016-07-10', 66, 'alternative'),
 ('2016-07-17', 84, 'alternative'),
 ('2016-07-24', 74, 'alternative'),
 ('2016-07-31', 71, 'alternative'),
 ('2016-08-07', 65, 'alternative'),
 ('2016-08-14', 68, 'alternative'),
 ('2016-08-21', 70, 'alternative'),
 ('2016-08-28', 78, 'alternative'),
 ('2016-09-04', 72, 'alternative'),
 ('2016-09-11', 73, 'alternative'),
 ('2016-09-18', 69, 'alternative'),
 ('2016-09-25', 66, 'alternative'),
 ('2016-10-02', 64, 'alternative'),
 ('2016-10-09', 67, 'alternative'),
 ('2016-10-16', 70, 'alternative'),
 ('2016-10-23', 73, 'alternative'),
 ('2016-10-30', 65, 'alternative'),
 ('2016-11-06', 64, 'alternative'),
 ('2016-11-13', 73, 'alternative'),
 ('2016-11-20', 70, 'alternative'),
 ('2016-11-27', 76, 'alternative'),
 ('2016-12-04', 70, 'alternative'),
 ('2016-12-11', 72, 'alternative'),
 ('2016-12-18', 78, 'alternative'),
 ('2016-12-25', 75, 'alternative'),
 ('2017-01-01', 70, 'alternative'),
 ('2017-01-08', 77, 'alternative'),
 ('2017-01-15', 68, 'alternative'),
 ('2017-01-22', 82, 'alternative'),
 ('2017-01-29', 75, 'alternative'),
 ('2017-02-05', 76, 'alternative'),
 ('2017-02-12', 71, 'alternative'),
 ('2017-02-19', 75, 'alternative'),
 ('2017-02-26', 70, 'alternative'),
 ('2017-03-05', 71, 'alternative'),
 ('2017-03-12', 72, 'alternative'),
 ('2017-03-19', 78, 'alternative'),
 ('2017-03-26', 72, 'alternative'),
 ('2017-04-02', 74, 'alternative'),
 ('2017-04-09', 72, 'alternative'),
 ('2017-04-16', 72, 'alternative'),
 ('2017-04-23', 76, 'alternative'),
 ('2017-04-30', 70, 'alternative'),
 ('2017-05-07', 69, 'alternative'),
 ('2017-05-14', 69, 'alternative'),
 ('2017-05-21', 70, 'alternative'),
 ('2017-05-28', 70, 'alternative'),
 ('2017-06-04', 61, 'alternative'),
 ('2017-06-11', 66, 'alternative'),
 ('

To see change in interest overtime, I did a Period-Over-Period Analysis with the window function LAG(). This function calculated the change in interest per year for each music genre. I also grouped and ordered the data by year and genre. 

In [17]:
%%sql
SELECT
    LEFT(`date`,4) AS google_year, genre,
    SUM(interest) AS yearly_interest,
    LAG(sum(interest),1) OVER(PARTITION BY genre ORDER BY LEFT(`date`, 4)) AS previous_year_interest,
    (SUM(interest)
    - LAG(sum(interest),1) OVER(PARTITION BY genre ORDER BY LEFT(`date`, 4)))
    / LAG(sum(interest),1) OVER(PARTITION BY genre ORDER BY LEFT(`date`, 4))
    * 100 AS interest_growth
FROM google_interest_by_genre
WHERE `date` BETWEEN "2015-05-03" AND "2020-04-26"
GROUP BY google_year, genre
ORDER BY genre, google_year;

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
54 rows affected.


[('2015', 'alternative', Decimal('2912'), None, None),
 ('2016', 'alternative', Decimal('3837'), Decimal('2912'), Decimal('31.7651')),
 ('2017', 'alternative', Decimal('3723'), Decimal('3837'), Decimal('-2.9711')),
 ('2018', 'alternative', Decimal('3358'), Decimal('3723'), Decimal('-9.8039')),
 ('2019', 'alternative', Decimal('3079'), Decimal('3358'), Decimal('-8.3085')),
 ('2020', 'alternative', Decimal('1072'), Decimal('3079'), Decimal('-65.1835')),
 ('2015', 'folk', Decimal('2655'), None, None),
 ('2016', 'folk', Decimal('3852'), Decimal('2655'), Decimal('45.0847')),
 ('2017', 'folk', Decimal('3837'), Decimal('3852'), Decimal('-0.3894')),
 ('2018', 'folk', Decimal('3397'), Decimal('3837'), Decimal('-11.4673')),
 ('2019', 'folk', Decimal('3068'), Decimal('3397'), Decimal('-9.6850')),
 ('2020', 'folk', Decimal('905'), Decimal('3068'), Decimal('-70.5020')),
 ('2015', 'jazz', Decimal('2595'), None, None),
 ('2016', 'jazz', Decimal('3545'), Decimal('2595'), Decimal('36.6089')),
 ('2017', 'jazz', Decimal('3663'), Decimal('3545'), Decimal('3.3286')),
 ('2018', 'jazz', Decimal('3449'), Decimal('3663'), Decimal('-5.8422')),
 ('2019', 'jazz', Decimal('3454'), Decimal('3449'), Decimal('0.1450')),
 ('2020', 'jazz', Decimal('1041'), Decimal('3454'), Decimal('-69.8610')),
 ('2015', 'pop', Decimal('2296'), None, None),
 ('2016', 'pop', Decimal('3407'), Decimal('2296'), Decimal('48.3885')),
 ('2017', 'pop', Decimal('3084'), Decimal('3407'), Decimal('-9.4805')),
 ('2018', 'pop', Decimal('2669'), Decimal('3084'), Decimal('-13.4565')),
 ('2019', 'pop', Decimal('2676'), Decimal('2669'), Decimal('0.2623')),
 ('2020', 'pop', Decimal('791'), Decimal('2676'), Decimal('-70.4410')),
 ('2015', 'rap', Decimal('2773'), None, None),
 ('2016', 'rap', Decimal('4081'), Decimal('2773'), Decimal('47.1691')),
 ('2017', 'rap', Decimal('3913'), Decimal('4081'), Decimal('-4.1166')),
 ('2018', 'rap', Decimal('3180'), Decimal('3913'), Decimal('-18.7324')),
 ('2019', 'rap', Decimal('2757'), Decimal('3180'), Decimal('-13.3019')),
 ('2020', 'rap', Decimal('841'), Decimal('2757'), Decimal('-69.4958')),
 ('2015', 'reggae', Decimal('3082'), None, None),
 ('2016', 'reggae', Decimal('4055'), Decimal('3082'), Decimal('31.5704')),
 ('2017', 'reggae', Decimal('3532'), Decimal('4055'), Decimal('-12.8977')),
 ('2018', 'reggae', Decimal('3099'), Decimal('3532'), Decimal('-12.2593')),
 ('2019', 'reggae', Decimal('3004'), Decimal('3099'), Decimal('-3.0655')),
 ('2020', 'reggae', Decimal('773'), Decimal('3004'), Decimal('-74.2676')),
 ('2015', 'rnb', Decimal('2998'), None, None),
 ('2016', 'rnb', Decimal('4129'), Decimal('2998'), Decimal('37.7252')),
 ('2017', 'rnb', Decimal('4112'), Decimal('4129'), Decimal('-0.4117')),
 ('2018', 'rnb', Decimal('3716'), Decimal('4112'), Decimal('-9.6304')),
 ('2019', 'rnb', Decimal('3559'), Decimal('3716'), Decimal('-4.2250')),
 ('2020', 'rnb', Decimal('1180'), Decimal('3559'), Decimal('-66.8446')),
 ('2015', 'rock', Decimal('2824'), None, None),
 ('2016', 'rock', Decimal('3756'), Decimal('2824'), Decimal('33.0028')),
 ('2017', 'rock', Decimal('3625'), Decimal('3756'), Decimal('-3.4878')),
 ('2018', 'rock', Decimal('3139'), Decimal('3625'), Decimal('-13.4069')),
 ('2019', 'rock', Decimal('3023'), Decimal('3139'), Decimal('-3.6954')),
 ('2020', 'rock', Decimal('889'), Decimal('3023'), Decimal('-70.5921')),
 ('2015', 'soul', Decimal('846'), None, None),
 ('2016', 'soul', Decimal('1437'), Decimal('846'), Decimal('69.8582')),
 ('2017', 'soul', Decimal('1011'), Decimal('1437'), Decimal('-29.6451')),
 ('2018', 'soul', Decimal('865'), Decimal('1011'), Decimal('-14.4411')),
 ('2019', 'soul', Decimal('798'), Decimal('865'), Decimal('-7.7457')),
 ('2020', 'soul', Decimal('270'), Decimal('798'), Decimal('-66.1654'))]

From the data, we see that interest in Alternative Music, Folk Music, Instrumental, Pop, Rap, Reggae, Rnb, Rock and Soul was decreasing progressively over 5 years

Jazz Music saw an increase from 2016 to 2017, a decrease from 2017-2018 and then held steady. 

Looking further into Jazz, was there a particular month that caused this spike? If so, can we then dig deeper to know if there was an event or album released that month in Jazz? 

I looked into the day to day data for Jazz and saw that on April 22, 2018, where interest was ranked 100, there was a Utah Jazz Basketball game where they beat Oklahoma City thunder 115-102. Although I had limited my API to the category of music, this would make sense why there was such a surge in interest for the word "jazz".  

While this complicated function seems insightful and paints an interesting story on the data, the results serve as a reminder of how difficult pulling accurate, targeted data can be. In the future, I would love to perform this query on a better data set, perhaps one provided by the company itself.



# Exploratory Query 2

I wanted to look into individual countries and see what music genre was the most popular in each of the countries. 

In order to do this, I used a CASE statement to categorize and label what genre was most popular in comparison to another in a country. 

In [23]:
%%sql
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS most_popular_genre_per_country
FROM popularity_by_country;


 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
250 rows affected.


[('Afghanistan', 54, 27, 4, 15, 0, 'Rock'),
 ('Albania', 47, 32, 5, 16, 0, 'Rock'),
 ('Algeria', 21, 20, 7, 50, 2, 'Rap'),
 ('American Samoa', 54, 16, 10, 20, 0, 'Rock'),
 ('Andorra', 57, 23, 3, 17, 0, 'Rock'),
 ('Angola', 9, 7, 49, 35, 0, 'Instrumental'),
 ('Anguilla', 52, 24, 10, 14, 0, 'Rock'),
 ('Antarctica', 0, 0, 0, 0, 0, 'none'),
 ('Antigua & Barbuda', 51, 25, 9, 15, 0, 'Rock'),
 ('Argentina', 50, 26, 4, 20, 0, 'Rock'),
 ('Armenia', 34, 36, 10, 20, 0, 'Pop'),
 ('Aruba', 57, 26, 6, 11, 0, 'Rock'),
 ('Australia', 57, 29, 2, 9, 3, 'Rock'),
 ('Austria', 68, 21, 2, 9, 0, 'Rock'),
 ('Azerbaijan', 27, 37, 11, 25, 0, 'Pop'),
 ('Bahamas', 54, 20, 7, 18, 1, 'Rock'),
 ('Bahrain', 49, 29, 6, 16, 0, 'Rock'),
 ('Bangladesh', 24, 15, 7, 54, 0, 'Rap'),
 ('Barbados', 61, 20, 7, 12, 0, 'Rock'),
 ('Belarus', 52, 27, 8, 12, 1, 'Rock'),
 ('Belgium', 51, 33, 2, 14, 0, 'Rock'),
 ('Belize', 52, 23, 9, 16, 0, 'Rock'),
 ('Benin', 17, 14, 30, 36, 3, 'Rap'),
 ('Bermuda', 62, 22, 5, 11, 0, 'Rock'),
 ('Bhutan', 27, 21, 4, 48, 0, 'Rap'),
 ('Bolivia', 29, 18, 7, 46, 0, 'Rap'),
 ('Bosnia & Herzegovina', 47, 39, 4, 10, 0, 'Rock'),
 ('Botswana', 40, 19, 16, 16, 9, 'Rock'),
 ('Bouvet Island', 0, 0, 0, 0, 0, 'none'),
 ('Brazil', 50, 27, 4, 19, 0, 'Rock'),
 ('British Indian Ocean Territory', 100, 0, 0, 0, 0, 'Rock'),
 ('British Virgin Islands', 69, 16, 8, 7, 0, 'Rock'),
 ('Brunei', 49, 35, 5, 11, 0, 'Rock'),
 ('Bulgaria', 53, 29, 5, 13, 0, 'Rock'),
 ('Burkina Faso', 39, 10, 11, 38, 2, 'Rock'),
 ('Burundi', 26, 19, 40, 15, 0, 'Instrumental'),
 ('Cambodia', 44, 36, 3, 17, 0, 'Rock'),
 ('Cameroon', 21, 17, 21, 38, 3, 'Rap'),
 ('Canada', 62, 26, 2, 10, 0, 'Rock'),
 ('Cape Verde', 29, 19, 21, 29, 2, 'none'),
 ('Caribbean Netherlands', 53, 33, 14, 0, 0, 'Rock'),
 ('Cayman Islands', 64, 21, 5, 10, 0, 'Rock'),
 ('Central African Republic', 44, 0, 20, 36, 0, 'Rock'),
 ('Chad', 20, 15, 18, 47, 0, 'Rap'),
 ('Chile', 44, 31, 5, 20, 0, 'Rock'),
 ('China', 43, 38, 5, 13, 1, 'Rock'),
 ('Christmas Island', 0, 0, 0, 0, 0, 'none'),
 ('Cocos (Keeling) Islands', 0, 0, 0, 0, 0, 'none'),
 ('Colombia', 37, 23, 15, 25, 0, 'Rock'),
 ('Comoros', 0, 0, 26, 74, 0, 'Rap'),
 ('Congo - Brazzaville', 13, 8, 32, 45, 2, 'Rap'),
 ('Congo - Kinshasa', 17, 12, 27, 41, 3, 'Rap'),
 ('Cook Islands', 61, 39, 0, 0, 0, 'Rock'),
 ('Costa Rica', 42, 25, 16, 17, 0, 'Rock'),
 ('Croatia', 55, 31, 3, 10, 1, 'Rock'),
 ('Cuba', 36, 27, 14, 23, 0, 'Rock'),
 ('Curaçao', 51, 28, 9, 12, 0, 'Rock'),
 ('Cyprus', 49, 37, 3, 9, 2, 'Rock'),
 ('Czechia', 72, 18, 2, 8, 0, 'Rock'),
 ('Côte d’Ivoire', 13, 16, 17, 52, 2, 'Rap'),
 ('Denmark', 57, 26, 2, 15, 0, 'Rock'),
 ('Djibouti', 46, 18, 3, 33, 0, 'Rock'),
 ('Dominica', 43, 28, 12, 17, 0, 'Rock'),
 ('Dominican Republic', 36, 21, 14, 29, 0, 'Rock'),
 ('Ecuador', 29, 15, 12, 44, 0, 'Rap'),
 ('Egypt', 50, 31, 4, 15, 0, 'Rock'),
 ('El Salvador', 29, 24, 12, 35, 0, 'Rap'),
 ('Equatorial Guinea', 11, 10, 32, 47, 0, 'Rap'),
 ('Eritrea', 100, 0, 0, 0, 0, 'Rock'),
 ('Estonia', 60, 27, 3, 10, 0, 'Rock'),
 ('Eswatini', 32, 17, 26, 17, 8, 'Rock'),
 ('Ethiopia', 40, 20, 16, 24, 0, 'Rock'),
 ('Falkland Islands (Islas Malvinas)', 72, 28, 0, 0, 0, 'Rock'),
 ('Faroe Islands', 60, 25, 2, 13, 0, 'Rock'),
 ('Fiji', 53, 24, 10, 13, 0, 'Rock'),
 ('Finland', 49, 44, 1, 6, 0, 'Rock'),
 ('France', 39, 33, 2, 25, 1, 'Rock'),
 ('French Guiana', 30, 27, 9, 31, 3, 'Rap'),
 ('French Polynesia', 46, 30, 5, 18, 1, 'Rock'),
 ('French Southern Territories', 0, 0, 0, 0, 0, 'none'),
 ('Gabon', 15, 12, 22, 48, 3, 'Rap'),
 ('Gambia', 26, 24, 26, 20, 4, 'none'),
 ('Georgia', 50, 31, 5, 14, 0, 'Rock'),
 ('Germany', 64, 21, 3, 12, 0, 'Rock'),
 ('Ghana', 27, 19, 38, 14, 2, 'Instrumental'),
 ('Gibraltar', 79, 16, 1, 4, 0, 'Rock'),
 ('Greece', 56, 31, 2, 9, 2, 'Rock'),
 ('Greenland', 57, 25, 0, 18, 0, 'Rock'),
 ('Grenada', 49, 26, 6, 17, 2, 'Rock'),
 ('Guadeloupe', 33, 27, 10, 28, 2, 'Rock'),
 ('Guam', 56, 28, 3, 13, 0, 'Rock'),
 ('Guatemala', 30, 18, 18, 34, 0, 'Rap'),
 ('Guernsey', 69, 23, 1, 7, 0, 'Rock'),
 (

The following results allowed us to view one of the 150 listed countries, and then see which genre of music was the most popular.

As an analyst, I could show this data to the Music Business Development team so that they can have an idea of what genre of music appeals to each of these 250 countries. Since Facebook has users in nearly every single country, this data can help the partnership team target specific music labels and partners to promote specifically to users in that country. 


I then wanted to understand which genre of music was most popular all together based on the data of these countries. So I created a Common Table Expression to hold the results of our Case query, and then performed a count on the most_popular_genre_per_country column, using a WHERE statement to limit genre. 

In [27]:
%%sql
WITH genre_ranking AS (
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS genre_ranking_by_country
FROM popularity_by_country
)
SELECT COUNT(*)
FROM genre_ranking
WHERE genre_ranking_by_country = "Rock"

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[(170,)]

The following query shows that the genre "Rock" is most popular in 170 countries. We will perform the same query to get a count for "Pop", "Instrumental", "Rap", "RnB", and "None" .

In [28]:
%%sql
WITH genre_ranking AS (
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS genre_ranking_by_country
FROM popularity_by_country
)
SELECT COUNT(*)
FROM genre_ranking
WHERE genre_ranking_by_country = "Pop"

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[(9,)]

In [29]:
%%sql
WITH genre_ranking AS (
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS genre_ranking_by_country
FROM popularity_by_country
)
SELECT COUNT(*)
FROM genre_ranking
WHERE genre_ranking_by_country = "Instrumental"

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[(10,)]

In [30]:
%%sql
WITH genre_ranking AS (
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS genre_ranking_by_country
FROM popularity_by_country
)
SELECT COUNT(*)
FROM genre_ranking
WHERE genre_ranking_by_country = "Rap"

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[(38,)]

In [31]:
%%sql
WITH genre_ranking AS (
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS genre_ranking_by_country
FROM popularity_by_country
)
SELECT COUNT(*)
FROM genre_ranking
WHERE genre_ranking_by_country = "RnB"

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[(0,)]

In [34]:
%%sql
WITH genre_ranking AS (
SELECT geoName, Rock, Pop, Instrumental, Rap, RnB,
    CASE
        WHEN Rock>Pop AND Rock>Instrumental AND Rock>Rap AND Rock>RnB THEN "Rock"
        WHEN Pop>Rock AND Pop>Instrumental AND Pop>Rap AND Pop>RnB THEN "Pop"
        WHEN Instrumental>Rock AND Instrumental>Pop AND Instrumental>Rap AND Instrumental>RnB THEN "Instrumental"
        WHEN Rap>Rock AND Rap>Pop AND Rap>Instrumental AND Rap>RnB THEN "Rap"
        WHEN RnB>Rock AND RnB>Pop AND RnB>Instrumental AND RnB>Rap THEN "RnB"
        ELSE "none"
    END AS genre_ranking_by_country
FROM popularity_by_country
)
SELECT COUNT(*)
FROM genre_ranking
WHERE genre_ranking_by_country = "None"

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[(23,)]

The results conclude that Rock is the genre with most interest on Google Trends, with 170 countries ranking it as their top genre. Then Rap with 38, Instrumental with 10, Pop with 9, and RnB with 0. 

As an analyst sending this back to Music Business Development team, I would be able to use this data to show what genre most broadly appeals to a diverse range of people. Rock, in this case, seems to be universally popular and gets many searches. Let's imagine that the business development team is asked by the marketing team to provide ideas of what music to use in the back of a global commercial. They can refer to these results, or more specific monthly results, to see what is trending and will resonate best in a commercial. 

# Exploratory Query 3

Can we find what genre of music various age groups are most interested in?

To do so, we will use a subquery to compare each genre against the top genre for each age group. 


In [36]:
%%sql
SELECT genre AS favorite_genre, max(16_19) AS percent_of_15_19_year_olds
FROM genre_by_age_group	
WHERE 16_19 = 
	(
		SELECT MAX(16_19)
		FROM genre_by_age_group);

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Pop Music', 52)]

In [37]:
%%sql
SELECT genre AS favorite_genre, max(20_24) AS percent_of_20_24_year_olds
FROM genre_by_age_group	
WHERE 20_24 = 
	(
		SELECT MAX(20_24)
		FROM genre_by_age_group);

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Pop Music', 54)]

In [38]:
%%sql
SELECT genre AS favorite_genre, max(25_34) AS percent_of_25_34_year_olds
FROM genre_by_age_group	
WHERE 25_34 = 
	(
		SELECT MAX(25_34)
		FROM genre_by_age_group);


 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Pop Music', 56)]

In [39]:
%%sql
SELECT genre AS favorite_genre, max(35_44) AS percent_of_35_44_year_olds
FROM genre_by_age_group	
WHERE 35_44 = 
	(
		SELECT MAX(35_44)
		FROM genre_by_age_group);

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Pop Music', 46)]

In [40]:
%%sql
SELECT genre AS favorite_genre, max(45_54) AS percent_of_45_54_year_olds
FROM genre_by_age_group	
WHERE 45_54 = 
	(
		SELECT MAX(45_54)
		FROM genre_by_age_group);

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Classic Rock', 62)]

In [41]:
%%sql
SELECT genre AS favorite_genre, max(55_64) AS percent_of_55_64_year_olds
FROM genre_by_age_group	
WHERE 55_64 = 
	(
		SELECT MAX(55_64)
		FROM genre_by_age_group);

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Classic Rock', 60)]

In [42]:
%%sql

SELECT genre AS favorite_genre, max(65_up) AS percent_of_65_up_year_olds
FROM genre_by_age_group	
WHERE 65_up = 
	(
		SELECT MAX(65_up)
		FROM genre_by_age_group);

 * mysql://admin:***@lmu-sql.c8zmikzpsjej.us-east-2.rds.amazonaws.com/sql_project
1 rows affected.


[('Classic Rock', 48)]

After running this query on all 7 age groups, the results are as follows:
Pop is the most popular genre for age groups between 16 year old and 44. 
Classic Rock is the most popular genre for age groups 45-65 and up. 

Passing this data along the the Music Business Development Team will support them in their search for partnerships. They may consider using pop music advertisements or pop influencers for their younger demographic. They can use pop artists to market to younger users. The same applies to their older audience who prefers rock music. Additionally, the older audience had country/ western music as their second preferred genre, so using pop music ads or partnerships to appeal to these users may not be a smart move. 